pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -861,9 +861,11 @@ cdef class _Timedelta(timedelta):
 
     def total_seconds(self):
         """
-        Total duration of timedelta in seconds (to ns precision).
+        Total duration of timedelta in seconds (to microsecond precision).
         """
-        return self.value / 1e9
+        # GH 31043
+        # Round down to microseconds to avoid confusing tzinfo.utcoffset
+        return (self.value - self.value % 1000) / 1e9
```

Comment:
We can also fix the bug with this:
```
return int(self.value) / 1000000000
```
By casting from `numpy.int64` to `int` and spelling out the denominator as an `int`, we switch to a different arithmetic implementation which handles long arithmetic better. This introduces an additional cast though and relies on the inner implementation of python. In addition, we end up returning a non-numpy type which clashes with the rest of our code.

Reply:
totally fine with what you have; i think we actually do this in a couple of places already

Reply:
Hmm. Something fishy is going on with the tests.

Reply:
k ping when ready